Spring-05

Max-Pochepets opened the pull request:
REST

- Created DTO for CinemaHall, Movie and MovieSession
- Created controller for respective model

Max-Pochepets committed:
first

Max-Pochepets committed:
Initial

Max-Pochepets committed:
Refactor:

changed format of MovieSessionResponseMapper's show time
created a Postman requests collection

Max-Pochepets committed:
Refactor:

changed get() in CinemaHallService and MovieService

Max-Pochepets committed:
Refactor:

get() in CinemaHall and Movie DAO's simplified

Max-Pochepets committed:
removed model from packages being scanned

Max-Pochepets committed:
Fix:

added Id field to MovieSessionRequestDto

Vedroid commented on src/main/java/cinema/shop/service/impl/CinemaHallServiceImpl.java:
```diff
@@ -19,6 +19,12 @@ public CinemaHall add(CinemaHall cinemaHall) {
+    @Override
+    public CinemaHall get(Long id) {
+        return cinemaHallDao.get(id).orElseThrow(()
+                -> new RuntimeException("There is no such cinema hall"));
```
RuntimeException - Less informative than NoSuchElementException

boroda4436 replied:
agree

boroda4436 commented on src/main/java/cinema/shop/controller/MovieSessionController.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    @DeleteMapping
+    public void delete(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
we should receive an Id of a resource we want to delete
See checklist's link. Try to find an example of `HTTP DELETE http://api.example.com/device-management/managed-devices`

boroda4436 commented on src/main/java/cinema/shop/model/dto/request/MovieSessionRequestDto.java:
```diff
@@ -0,0 +1,40 @@
+package cinema.shop.model.dto.request;
+
+public class MovieSessionRequestDto {
+    private Long id;
```
How we can use it?

Max-Pochepets committed:
Fix:

removed Id from MovieSessionRequestDto
added path variables to MovieSessionController's update and delete
changed the exception being thrown from getting object out of Optional from Runtime to NoSuchElement

Max-Pochepets committed:
checkstyle

boroda4436 commented on src/main/java/cinema/shop/controller/MovieSessionController.java:
```diff
@@ -0,0 +1,73 @@
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id,
+                       @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Do you still need it?

Max-Pochepets committed:
Fix:

changed MovieSessionController logic
chain of changes goes down until the DAO layer
